### PR TITLE
DPath Revamp - Logic

### PR DESCRIPTION
**Pull request by blurpesec**
### Description
Revamps DPath logic to scan for balances across actively-used dpaths. Splits New DPath scanner to use wallet type `LEDGER_NANO_S_NEW` so we can display both types in dev mode for now. Will overwrite back to `LEDGER_NANO_S` when dpath scanner is done c:

ToDos:
- [x] Right now this only will show you addresses and won't let you continue on to add all of them to dashboard. Add in radio buttons and onUnlock for next steps in AddAccount flow.
- [x] Speed up scanning. Extract out balance lookup to be done in parallel
- [x] Support scanning by a different token.
- [x] U2F & USB Ledger support.
- [x] Mnemonic support.
- [x] Trezor support.
- [x] Hardened Derivation scanning.
- [x] Use DPaths scanning for each device type.
- [x] DPath offset system.
- [x] Secondary scan.

Then: 
- [x] Test trezor scanning & adding.
- [x] Test ledger scanning & adding.
- [x] Test mnemonic scanning & adding.

Future PRs:
- [ ] Custom DPaths yo.
- [ ] Support generating empty addresses.
- [ ] Bluetooth Ledger support.

### CODE REVIEW

**Comment by blurpesec** [comment]
[ch5949]

File: shared/enclave/server/wallets/ledger.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ const Ledger: WalletLib = {
       const res = await app.getAddress(dpath, false, true);
       return {
         publicKey: res.publicKey,
-        chainCode: res.chainCode
+        chainCode: res.chainCode as string // @todo - figure out the cause of this `string | undefined`
```
Comment (Mrtenz) [inline shared/enclave/server/wallets/ledger.ts:37]: `getAddress` is declared as
```ts
    public getAddress(
      path: string,
      display?: boolean,
      chainCode?: boolean
    ): Promise<{ publicKey: string; address: string; chainCode?: string }>;
```

If you don't want to cast it to a string manually, you can declare `getAddress` as:
```ts
    getAddress(path: string, display?: boolean, chainCode?: false): Promise<{ publicKey: string; address: string }>;
    getAddress(
      path: string,
      display?: boolean,
      chainCode?: true
    ): Promise<{ publicKey: string; address: string; chainCode: string }>;
```

File: src/config/dpaths.ts [hunk 2]
```diff
@@ -238,3 +246,23 @@ export const DPathsList = {
 // ie. "m / 44' / 0' / 0'" is valid, "m / 4 4' / 0' / 0'" is invalid
 export const dPathRegex = /m\/4[4,9]'\/[0-9]+\'\/[0-9]+(\'+$|\'+(\/[0-1]+$))/;
 // const whitespaceDPathRegex = /m\s*\/\s*44'\s*\/\s*[0-9]+\'\s*\/\s*[0-9]+(\'+$|\'+\s*(\/\s*[0-1]+$))/;
+
+/**
+ * Due to limitations in the Ledger ETH application, only derivation paths starting with
+ * `m/44'/60'` and `m/44'/1'` can be checked.
+ */
```
Comment (Mrtenz) [inline src/config/dpaths.ts:253]: This only applies to Ethereum, not to other networks that Ledger Live supports.

**Comment by GuillaumeRx** [comment]
Don't forget to rebase 🙃

File: src/components/WalletUnlock/DeterministicAccountList.tsx [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+import BN from 'bn.js';
+import uniqBy from 'ramda/src/uniqBy';
+import prop from 'ramda/src/prop';
+
+import { translateRaw } from '@translations';
+import { ExtendedAsset, TAddress } from '@types';
+import { EthAddress, FixedSizeCollapsibleTable, Spinner, Checkbox, Button } from '@components';
+import { truncate, isSameAddress } from '@utils';
+import { BREAK_POINTS, SPACING, breakpointToNumber } from '@theme';
+import { DWAccountDisplay } from '@services/WalletService/deterministic/types';
```
Comment (GuillaumeRx) [inline src/components/WalletUnlock/DeterministicAccountList.tsx:12]: Can those imports be only from the alias ?

**Comment by blurpesec** [comment]
Will do this morning. Will also remove the staging testing stuff from production release

File: src/components/WalletUnlock/Ledger.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+import translate, { translateRaw, Trans } from '@translations';
+import { NewTabLink, Spinner, Button, DeterministicAccountList, AssetDropdown } from '@components';
+import {
+  EXT_URLS,
+  LEDGER_DERIVATION_PATHS,
+  DEFAULT_NUM_OF_ACCOUNTS_TO_SCAN,
+  DEFAULT_GAP_TO_SCAN_FOR
+} from '@config';
+import { NetworkContext, getNetworkById, getAssetByUUID, AssetContext } from '@services/Store';
+import { useDeterministicWallet } from '@services/WalletService';
```
Comment (GuillaumeRx) [inline src/components/WalletUnlock/Ledger.tsx:16]: Same here

File: src/components/WalletUnlock/NewMnemonic.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+import {
+  DeterministicAccountList,
+  AssetDropdown,
+  TogglablePassword,
+  Tooltip,
+  Input
+} from '@components';
+import { DPathsList, DEFAULT_NUM_OF_ACCOUNTS_TO_SCAN, DEFAULT_GAP_TO_SCAN_FOR } from '@config';
+import { NetworkContext, getNetworkById, getAssetByUUID, AssetContext } from '@services/Store';
+import { useDeterministicWallet } from '@services/WalletService';
```
Comment (GuillaumeRx) [inline src/components/WalletUnlock/NewMnemonic.tsx:18]: Same here

File: src/components/WalletUnlock/NewTrezor.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+import translate, { translateRaw } from '@translations';
+import { Spinner, Button, DeterministicAccountList, AssetDropdown } from '@components';
+import {
+  EXT_URLS,
+  TREZOR_DERIVATION_PATHS,
+  DEFAULT_NUM_OF_ACCOUNTS_TO_SCAN,
+  DEFAULT_GAP_TO_SCAN_FOR
+} from '@config';
+import { NetworkContext, getNetworkById, getAssetByUUID, AssetContext } from '@services/Store';
+import { useDeterministicWallet } from '@services/WalletService';
```
Comment (GuillaumeRx) [inline src/components/WalletUnlock/NewTrezor.tsx:16]: Same here

**Pull request merged**